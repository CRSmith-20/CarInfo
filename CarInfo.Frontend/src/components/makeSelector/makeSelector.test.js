import {shallow, mount} from 'enzyme';
import MakeSelector from './makeSelector';
import axios from "axios";
import { BrowserRouter as Router } from 'react-router-dom';
import { act } from "react-dom/test-utils";
import toJson from 'enzyme-to-json';

jest.mock("axios");


describe('<MakeSelector />', () => {
    describe('when component initalizes', () => {
        it('should initalize to loading', () => {
            const test = shallow(<Router><MakeSelector /></Router>);
            expect(toJson(test)).toMatchSnapshot();
        });
    });
    
    describe('when component loads', () => {
        it('should render makes with images', async () => {
            const makes = [
                "Hyundai", "Ford"
            ]
            
            let wrapper;
            
            await act(async () => {
                axios.get.mockResolvedValueOnce({data: makes});
                wrapper = mount(<Router><MakeSelector /></Router>);
            });
    
            wrapper.update();

            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find('Link').length).toBe(2);
        })
    });
    
    describe('when component loads', () => {
        it('should render makes with images', async () => {          
            let wrapper;
            
            await act(async () => {
                axios.get.mockRejectedValueOnce(new Error("Generic Failure"));
                wrapper = mount(<Router><MakeSelector /></Router>);
            });
    
            wrapper.update();

            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find('ErrorDisplay').length).toBe(1);
        })
    });
})