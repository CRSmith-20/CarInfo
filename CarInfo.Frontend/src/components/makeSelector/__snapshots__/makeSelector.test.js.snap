// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MakeSelector /> when component initalizes should initalize to loading 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MakeSelector />
</Router>
`;

exports[`<MakeSelector /> when component loads should render makes with images 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MakeSelector>
      <div>
        <h1
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Select Make
        </h1>
        <div
          key="Hyundai"
          style={
            Object {
              "display": "inline-block",
              "maxHeight": "500px",
              "maxWidth": "500px",
            }
          }
        >
          <Link
            to="/Hyundai"
          >
            <LinkAnchor
              href="/Hyundai"
              navigate={[Function]}
            >
              <a
                href="/Hyundai"
                onClick={[Function]}
              >
                <img
                  src="/MakerImages/Hyundai.png"
                  style={
                    Object {
                      "maxHeight": "500px",
                      "maxWidth": "500px",
                    }
                  }
                />
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <div
          key="Ford"
          style={
            Object {
              "display": "inline-block",
              "maxHeight": "500px",
              "maxWidth": "500px",
            }
          }
        >
          <Link
            to="/Ford"
          >
            <LinkAnchor
              href="/Ford"
              navigate={[Function]}
            >
              <a
                href="/Ford"
                onClick={[Function]}
              >
                <img
                  src="/MakerImages/Ford.png"
                  style={
                    Object {
                      "maxHeight": "500px",
                      "maxWidth": "500px",
                    }
                  }
                />
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </div>
    </MakeSelector>
  </Router>
</BrowserRouter>
`;

exports[`<MakeSelector /> when component loads should render makes with images 2`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MakeSelector>
      <ErrorDisplay
        title="Select Make"
      >
        <div>
          <h1
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Select Make
          </h1>
          <Alert
            closeLabel="Close alert"
            key="error"
            show={true}
            style={
              Object {
                "textAlign": "center",
              }
            }
            transition={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "appear": false,
                  "in": false,
                  "mountOnEnter": false,
                  "timeout": 300,
                  "unmountOnExit": false,
                },
                "render": [Function],
              }
            }
            variant="warning"
          >
            <Fade
              appear={false}
              in={true}
              mountOnEnter={false}
              style={
                Object {
                  "textAlign": "center",
                }
              }
              timeout={300}
              unmountOnExit={true}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
                timeout={300}
                unmountOnExit={true}
              >
                <div
                  className="fade alert alert-warning show"
                  role="alert"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  An error has occurred while loading, please refresh and try again.
                </div>
              </Transition>
            </Fade>
          </Alert>
        </div>
      </ErrorDisplay>
    </MakeSelector>
  </Router>
</BrowserRouter>
`;
